AutoCollection/HttpDependencyParser.ts

```diff
@@ -139,7 +139,7 @@ class HttpDependencyParser extends RequestParser {
         }
 
         // Mix in default values used by http.request and others
-        options.protocol = options.protocol || (<any>request).agent.protocol;
+        options.protocol = options.protocol || ((<any>request).agent && (<any>request).agent.protocol) || undefined;
```

Comment:
Just to be clear, this doesn't do anything about `protocol` being undefined right? But rather if `agent` doesn't exist?

Reply:
That’s right